jupyter_client/jsonutil.py

```diff
@@ -108,5 +109,14 @@ def json_default(obj):
         return obj.isoformat().replace('+00:00', 'Z')
     elif isinstance(obj, bytes):
         return b2a_base64(obj).decode('ascii')
+    elif (
+        isinstance(obj, (set, types.GeneratorType)) or
+        hasattr(obj, '__iter__') and hasattr(obj, '__next__')
+    ):
+        return list(obj)
+    elif hasattr(obj, '__int__'):
+        return int(obj)
```

Comment:
A `float` also has a `__int__` method, so this will convert all floats to ints?

Reply:
The test for floats is passing so I don't believe so

Reply:
But I changed the check anyway

Reply:
Thanks for the review!